vendor/bat-native-ads/src/bat/ads/internal/bundle.cc

```diff
@@ -122,15 +117,35 @@ std::unique_ptr<BundleState> Bundle::GenerateFromCatalog(
       // Ad notification creatives
       for (const auto& creative : creative_set.creative_ad_notifications) {
         if (!DoesOsSupportCreativeSet(creative_set)) {
+          BLOG(1, "creative set id " << creative_set.creative_set_id
+              << " not available for " << GetClientOS());
+
           continue;
         }
 
         CreativeAdNotificationInfo info;
         info.creative_instance_id = creative.creative_instance_id;
         info.creative_set_id = creative_set.creative_set_id;
         info.campaign_id = campaign.campaign_id;
-        info.start_at_timestamp = campaign.start_at;
-        info.end_at_timestamp = campaign.end_at;
+
+        base::Time start_at_time;
+        if (base::Time::FromUTCString(campaign.start_at.c_str(),
+            &start_at_time)) {
+          info.start_at_timestamp =
+              static_cast<int64_t>(start_at_time.ToDoubleT());
+        } else {
+          info.start_at_timestamp = std::numeric_limits<int64_t>::min();
+        }
+
+        base::Time end_at_time;
+        if (base::Time::FromUTCString(campaign.end_at.c_str(),
+            &end_at_time)) {
+          info.end_at_timestamp =
+              static_cast<int64_t>(end_at_time.ToDoubleT());
+        } else {
+          info.end_at_timestamp = std::numeric_limits<int64_t>::max();
```

Comment:
Is this worth logging? Are we essentially saying that if the end time is corrupted run it forever? I would've thought we'd just cap it at some arbitrary point in the future.

Reply:
Added logs